xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/Visualization.h

```diff
@@ -122,9 +159,170 @@ namespace kodi
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// #include <kodi/visualization/Visualization.h>
+  ///
+  /// class CMyVisualization : public ::kodi::addon::CInstanceVisualization
+  /// {
+  /// public:
+  ///   CMyVisualization(void* instance);
+  ///
+  ///   bool Start(int channels, int samplesPerSec, int bitsPerSample, std::string songName) override;
+  ///   void AudioData(const float* audioData, int audioDataLength, float* freqData, int freqDataLength) override;
+  ///   void Render() override;
+  /// };
+  ///
+  /// CMyVisualization::CMyVisualization(void* instance)
+  ///   : CInstanceVisualization(instance)
+  /// {
+  ///   ...
+  /// }
+  ///
+  /// bool CMyVisualization::Start(int channels, int samplesPerSec, int bitsPerSample, std::string songName)
+  /// {
+  ///   ...
+  ///   return true;
+  /// }
+  ///
+  /// void CMyVisualization::AudioData(const float* audioData, int audioDataLength, float* freqData, int freqDataLength)
+  /// {
+  ///   ...
+  /// }
+  ///
+  /// void CMyVisualization::Render()
+  /// {
+  ///   ...
+  /// }
+  ///
+  ///
+  /// /*----------------------------------------------------------------------*/
+  ///
+  /// class CMyAddon : public ::kodi::addon::CAddonBase
+  /// {
+  /// public:
+  ///   CMyAddon() { }
+  ///   ADDON_STATUS CreateInstance(int instanceType,
+  ///                               std::string instanceID,
+  ///                               KODI_HANDLE instance,
+  ///                               KODI_HANDLE& addonInstance) override;
+  /// };
+  ///
+  /// /* If you use only one instance in your add-on, can be instanceType and
+  ///  * instanceID ignored */
+  /// ADDON_STATUS CMyAddon::CreateInstance(int instanceType,
+  ///                                       std::string instanceID,
+  ///                                       KODI_HANDLE instance,
+  ///                                       KODI_HANDLE& addonInstance)
+  /// {
+  ///   if (instanceType == ADDON_INSTANCE_VISUALIZATION)
+  ///   {
+  ///     kodi::Log(ADDON_LOG_NOTICE, "Creating my Visualization");
+  ///     addonInstance = new CMyVisualization(instance);
+  ///     return ADDON_STATUS_OK;
+  ///   }
+  ///   else if (...)
+  ///   {
+  ///     ...
+  ///   }
+  ///   return ADDON_STATUS_UNKNOWN;
+  /// }
+  ///
+  /// ADDONCREATOR(CMyAddon)
+  /// ~~~~~~~~~~~~~
+  ///
+  /// The **desctruction** of the here created example class `CMyVisualization`
+  /// becomes done from Kodi's header and never for add-on instances a delete on
+  /// add-on itself necessary.
+  ///
+  //----------------------------------------------------------------------------
   class CInstanceVisualization : public IAddonInstance
   {
   public:
+    //==========================================================================
+    ///
+    /// @ingroup cpp_kodi_addon_visualization
+    /// @brief Visualization class constructor
+    ///
+    /// Used on addon who supports only visualization and CAddonBase is also
```

Comment:
Does this class inherit from `CAddonBase`? It looks like it only inherits from `IAddonInstance`.